rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -200,8 +200,17 @@ namespace ros {
         while( true )
         {
           int data = hardware_.read();
-          if( data < 0 )
+          if( data < 0 ) {
+#if defined(USE_WIFICON)
+		    if (!hardware_.isConnectionUp()) {		// check connection, if it is broken try to repair it N.B. should really add interrupt timer as this blocks !
+				configured_ = false;
+				return -3;
+			} else
+				break;
+#else
```

Comment:
Are you worried about code bloat or what?

The Wifi connection is fundamentally different than every single other instance of rosserial_arduino, as it doesn't use any serial (rs232) class, so can't get any of the nice hardware level managment.